go-controller/pkg/util/iptables.go

```diff
@@ -141,7 +141,20 @@ func (f *FakeIPTables) ClearChain(tableName, chainName string) error {
 // DeleteChain deletes the chain in the specified table.
 // The chain must be empty
 func (f *FakeIPTables) DeleteChain(tableName, chainName string) error {
-	return f.DeleteChain(tableName, chainName)
+	table, err := f.getTable(tableName)
+	if err != nil {
+		return err
+	}
+	if chain, err := table.getChain(chainName); err == nil {
+		if len(chain) != 0 {
+			return fmt.Errorf("chain must be empty")
+		}
+		// chain exists, flush the rules
+		delete((*table), chainName)
```

Comment:
@squeed Sorry, one small NIT. The comment is not right. There are no rules to flush since the chain should be empty. So, `chain exists, delete the chain`, perhaps?

Reply:
@girishmg good point; silly copy-and-paste. The code is clear; I removed the comment.